core/src/main/java/org/mule/runtime/core/internal/processor/TryScope.java

```diff
@@ -68,18 +68,16 @@ public CoreEvent process(final CoreEvent event) throws MuleException {
     ExecutionTemplate<CoreEvent> executionTemplate =
         createScopeTransactionalExecutionTemplate(muleContext, transactionConfig);
     ExecutionCallback<CoreEvent> processingCallback = () -> {
+      TransactionAdapter transaction = (TransactionAdapter) TransactionCoordination.getInstance().getTransaction();
+      ComponentLocation lastLocation = transaction.getComponentLocation().orElse(null);
+      transaction.setComponentLocation(getLocation());
+
       try {
-        TransactionAdapter transaction = (TransactionAdapter) TransactionCoordination.getInstance().getTransaction();
-        ComponentLocation lastLocation = transaction.getComponentLocation().orElse(null);
-        transaction.setComponentLocation(getLocation());
-        CoreEvent e = processToApply(event,
-                                     p -> from(p).flatMap(request -> processWithChildContext(request, nestedChain,
-                                                                                             ofNullable(getLocation()),
-                                                                                             messagingExceptionHandler)));
+        return processWithChildContextBlocking(event, nestedChain, ofNullable(getLocation()), messagingExceptionHandler);
+      } catch (Throwable t) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Catch Exception instead of Throwable. ([squid:S1181](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1181))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7683&issues=AWjyMA2RotIXnmsUSXGZ&open=AWjyMA2RotIXnmsUSXGZ)

[ISSUE_KEY]: AWjyMA2RotIXnmsUSXGZ